backend/src/middleware-callbacks/GroupUserCallbacks.ts

```diff
@@ -0,0 +1,148 @@
+    return value != null && value !== "" && !isNaN(Number(value.toString()));
+  };
+
+  public verifyUser = (
+    
+    request: Request<
+      {userId: string,groupId: string},
+      {},
+      { userAccountData: UserAccountData } & CreateGroupUserDto &
+        AddMemberGroupUserDto
+    >,
+    response: Response,
+    next: NextFunction
+  ) => {
+    console.log(request.params?.userId);
+    
+    return GroupUserService.verifyUser(parseInt(request.params?.userId), parseInt(request.params?.groupId));
+  };
+
+  public addGroupUserByUserEmail = async (
+    
+    request: Request<
+      {},
+      {},
+      { userAccountData: UserAccountData } & CreateGroupUserDto &
+        AddMemberGroupUserDto
+    >,
+    response: Response,
+    next: NextFunction
+  ) => {
+    const requesterId = request.body.userAccountData.userAccountId;
+    const groupId = request.body.groupId;
+
+
+    if (!this.isNumber(requesterId))
+      return response
+        .status(RESPONSE_STATUS.BAD_REQUEST)
+        .json({ message: "Missing the requester id or improper data type" });
+    return GroupService.findGroupById(groupId as unknown as number).then(
+      (foundGroup) => {
+        if (!foundGroup)
+          return response
+            .status(RESPONSE_STATUS.NOT_FOUND)
+            .json({ message: "Group not found " });
+        if (foundGroup.ownerId !== requesterId)
+          return response
+            .status(RESPONSE_STATUS.NOT_FOUND)
+            .json({ message: "Requester has no permission!" });
+        return GroupUserService.getGroupUserByGroupId(groupId)
+          .then((foundGroupCollection) => {
+            for (let index = 0; index < foundGroupCollection.length; index++) {
+              const element = foundGroupCollection[index];
+              
+              
+              const user = element.user;
+              const account = user.account;
+              if (account.email === request.body.email) {
+                if (element.membershipAccepted === true)
+                  return response.json({ message: "Already member!" });
+                else {
+                  console.log("here");
+                  console.log(account.email);
+
+                  transporter.sendMail({
+                    to: request.body.email,
+                    from: "257307@student.pwr.edu.pl", 
+                    subject: "Group trip invitation!",
+                    html:"Click to accept <a href='http://127.0.0.1:5000/groupUser/verify/" + element.userId+"/"+element.groupId+"'> invitation!</a>",
+                    });
+                  
+                  return response.json({message: "Ok!"});
+                }
+              }
+            }
+            return response.json({message: "Something went wrong!"})
+          })
+          .catch((error) => {
+            return response.status(500).json({ error });
+          });
+      }
+    );
+  };
+
+  public deleteGroupUserByUserId = async (
```

Comment:
delete user group ?? please add check of ownership

Reply:
fixed in the following pull request